### PR TITLE
Allow updating ENGINE_CONFIG for System.updatelogtables

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Allow you to update the ENGINE_CONFIG via System.updatelogtables if you are not changing the actual storage engine.
I needed to go from InnoDB `ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=4` to InnoDB `ROW_FORMAT=DYNAMIC` and you can't do it via this API function without changing the engine too.

Also return the number of tables we updated as a result instead of "1" in the API call.

Before
----------------------------------------
Cannot modify SQL engine config for log tables.

After
----------------------------------------
Can modify SQL engine config for log tables.

Technical Details
----------------------------------------
Just follows the same principal as existing code.  Enabled via parameter so no change by default.

Comments
----------------------------------------
@seamuslee001 @mfb @pfigel

### CODE REVIEW

**Comment by seamuslee001** [comment]
Test failure relates @mattwire https://test.civicrm.org/job/CiviCRM-Core-PR/26463/testReport/junit/(root)/CRM_Logging_SchemaTest/testColumnInfo/

**Comment by pfigel** [comment]
I'm wondering if we really need a new API parameter or if just checking for a difference is good enough. Seems like it would be a sane default behaviour, unless we have good reason to be more conservative. Thoughts?

**Comment by mattwire** [comment]
@pfigel I think it makes sense to be conservative (and add the param) when we first add it, just in case it should cause any problems.

**Review by pfigel — CHANGES_REQUESTED** [review]
@mattwire LGTM!

* General standards
    * ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) __PASS__
    * ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __PASS__: No change in default behaviour - hidden behind new parameter with appropriate docs.
    * ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __PASS__
    * ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) __PASS__: Confirmed engine config is changed with `updateChangedEngineConfig=1` and not changed otherwise.
* Developer standards
    * ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __PASS__: Minor (useful) change in return value, unlikely to have ecosystem impact.
    * ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __PASS__
    * ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __PASS__
    * ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __PASS__

**Review by pfigel — APPROVED** [review]
Whoops, noticed the test failure was already resolved, so definitely LGTM now.

**Comment by monishdeb** [comment]
@mattwire @pfigel is there anything remaining or I can merged it given that @pfigel has already reviewed and approved it?

**Comment by mattwire** [comment]
@monishdeb I'm happy!

**Comment by monishdeb** [comment]
Great .. Let us wait for @pfigel reply and I'll then merge this PR after his approval.

**Comment by pfigel** [comment]
@monishdeb I'm happy with this!

**Comment by monishdeb** [comment]
Merged. Thanks guys for all your effort.

**Pull request merged**
